### ISSUE TITLE
memory leak/crash

### ISSUE THREAD

**Issue by bugficks** [extension-js/extension.js#127]
hi,

every now and then `npm run dev` crashes or takes ages to build.

`
🧩 Extension.js ►►► testext (v1.0) (webpack 5.92.1) compiled successfully in 64661 ms
`

```
🧩 Extension.js ►►► testext (v1.0) (webpack 5.92.1) compiled successfully in 2473 ms

<--- Last few GCs --->

[89876:00000246DB880080] 52607804 ms: Mark-Compact 4048.8 (4136.8) -> 4037.3 (4140.8) MB, 3553.52 / 0.00 ms  (average mu = 0.457, current mu = 0.025) allocation failure; scavenge might not succeed
[89876:00000246DB880080] 52613244 ms: Mark-Compact 4053.4 (4141.0) -> 4041.7 (4145.3) MB, 5335.93 / 0.00 ms  (average mu = 0.242, current mu = 0.019) allocation failure; scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
----- Native stack trace -----

 1: 00007FF7039672EB node::SetCppgcReference+18123
 2: 00007FF7038D8F04 DSA_meth_get_flags+90724
 3: 00007FF704350701 v8::Isolate::ReportExternalAllocationLimitReached+65
 4: 00007FF704339E58 v8::Function::Experimental_IsNopFunction+1336
 5: 00007FF70419B8A0 v8::Platform::SystemClockTimeMillis+659328
 6: 00007FF704198928 v8::Platform::SystemClockTimeMillis+647176
 7: 00007FF7041ADC3A v8::Platform::SystemClockTimeMillis+733978
 8: 00007FF7041AE4B7 v8::Platform::SystemClockTimeMillis+736151
 9: 00007FF7041BCDAF v8::Platform::SystemClockTimeMillis+795791
10: 00007FF703E7CD05 v8::CodeEvent::GetFunctionName+116773
11: 00007FF70440208E v8::PropertyDescriptor::writable+678094
12: 00007FF7044027E3 v8::PropertyDescriptor::writable+679971
13: 00007FF6848ACEC5
```

**Comment by cezaraugusto** [comment]
hi @bugficks thanks for reporting!

Are you able to reproduce it consistently? or maybe some code/steps I can run locally so I can reproduce as well

**Comment by bugficks** [comment]
yes, if I let it run over night I will be greeted by this in the morning. 
I haven't noticed any pattern for the long build time issued, randomly happens through out the day.

**Comment by cezaraugusto** [comment]
That helps a lot, thanks. I'll keep it running overnight and try to catch it

**Comment by bugficks** [comment]
Yesterday it got that bad that it crashed or hung within less than 10 saves.
I have tried disabling 'reload service' and 'extension manager' just to see if it makes any difference, it didn't. 
So I added _launch.json_ to vscode using exactly same command line parameters, except those 2 extensions.
Running `npm run dev` worked without issues so far.

Now the kicker, this morning I have tried to reproduce the problem having visual studio attached but problem seems to be gone.
Same vscode instance, same terminal instance, same shell instance, same node version, same chrome version 🤔
I did do `npm update` but diff only shows `electron-to-chromium 1.4.825->1.4.827` 

Now I am puzzled to say the least

**Comment by cezaraugusto** [comment]
that's a tricky one. are you using the public/ folder? aside from webpack the only processes that persist during development are the browser instance and a listener to [`public/`](https://github.com/extension-js/extension.js/blob/b3b6f51fa18984574d1a53b32a573068e8d9cd02/programs/develop/webpack/plugins/SpecialFoldersPlugin/CopyPublicFolder.ts#L66). I'm not able to reproduce

**Comment by cezaraugusto** [comment]
btw thanks for all the bug reports, really helpful

**Comment by bugficks** [comment]
I have only one file in _public_, a small config.json.

**Comment by bugficks** [comment]
I have added '--node-memory-debug --inspect --max_old_space_size=1024' to _extension.cmd_ and waited for it to crash:

![image](https://github.com/user-attachments/assets/d21f1f61-8a8a-4b2c-a6bb-7a52f5187f08)
![image](https://github.com/user-attachments/assets/8c0b15ed-df92-4598-9dc4-8c3cf38f5f0d)


maybe this helps, to me it looks like "file change detection" might be the root cause of this.

**Comment by cezaraugusto** [comment]
hey @bugficks! thanks again. I'm rewriting the whole `develop` package, hopefully the bug will vanish along with it or provide us a better view of what's going on. I'm pretty sure it's related to watching the public/ folder via custom watcher

btw added you as a collaborator to the project so you can help me reviewing the upcoming update. feel free to report other issues as you find them or triage existing. the ultimate goal is to make it very easy to develop extensions.

**Comment by bugficks** [comment]
Thanks for invitation but I wont have the time required for it. I will continue to report bugs as I encounter them while working with this.

**Comment by bugficks** [comment]
I removed public folder, copied my config into dist dir manually and it still crashes. 
Anyways I've started to use webpack directly as I easy needed access to build steps and couldn't stand the wait/crashes anymore.
While porting I've encountered this:
![1](https://github.com/user-attachments/assets/2acc5956-77ab-41d6-9c49-a1d80ac7c5fc)
![2](https://github.com/user-attachments/assets/d4740da8-4490-4524-aef8-08deeb8d35b3)

**Comment by cezaraugusto** [comment]
great catch! that's a dependency of `glob` which we use. Will cut off that bad boy

**Comment by cezaraugusto** [comment]
@bugficks took me a while but finally fixed this one. might giving it another try?

**Comment by bugficks** [comment]
sure, if here an easy way to test it. 

tried installing from github but that didn't work:
![clipboard](https://github.com/user-attachments/assets/e32ba793-dd60-4cd1-a1b0-24de54e507a0)

**Comment by cezaraugusto** [comment]
It seems you are downloading v1.8.0. Please try with current latest, `pnpm add extension@2.0.0-alpha.28`

**Comment by bugficks** [comment]
Looks like it is fixed. No more crashes over night and builds are way faster too now.

**Comment by cezaraugusto** [comment]
I'm happy to know that! Going to close this one as fixed then. If you see soomething else wrong, let me know!
